src/connectors/WalletConnectConnector/index.ts

```diff
@@ -0,0 +1,146 @@
+import { UserRejectedRequestError } from '@pangolindex/web3-react-injected-connector';
+import WalletProvider, { EthereumProvider } from '@walletconnect/ethereum-provider';
+import { AbstractConnector } from '@web3-react/abstract-connector';
+import { ConnectorUpdate } from '@web3-react/types';
+
+export interface WalletConnectConnectorArguments {
+  rpcMap: { [chainId: number]: string };
+  projectId: string;
+  metadata: {
+    name: string;
+    description: string;
+    url: string;
+    icons: string[];
+    verifyUrl?: string;
+  };
+}
+
+export class WalletConnectConnector extends AbstractConnector {
+  public provider?: WalletProvider;
+  private config: WalletConnectConnectorArguments;
+
+  constructor(config: WalletConnectConnectorArguments) {
+    const supportedChainIds = Object.keys(config.rpcMap).map((k) => Number(k));
+    super({ supportedChainIds: supportedChainIds });
+    this.config = config;
+    this.handleChainChanged = this.handleChainChanged.bind(this);
+    this.handleAccountsChanged = this.handleAccountsChanged.bind(this);
+    this.handleDisconnect = this.handleDisconnect.bind(this);
+
+    if ((this.config.projectId, length > 0)) {
+      this.init();
+    }
+  }
+
+  private async init() {
+    const supportedChainIds = this.getSupportedChainIds();
+
+    const rpcMap = Object.entries(this.config.rpcMap).reduce((acc, [chainId, rpc]) => {
+      acc[`eip155:${chainId}`] = rpc;
```

Comment:
why do we need `eip155` here ?

Reply:
because walletconnect provider require this in chain key without this the provider throw an error

bifrost don't support avalanche because this is throwing this error

https://github.com/WalletConnect/walletconnect-monorepo/blob/6ab8b3583b9db07dbd2a3a9acf5580892a115c11/providers/ethereum-provider/src/EthereumProvider.ts#L210C31-L210C31

https://github.com/WalletConnect/walletconnect-monorepo/blob/6ab8b3583b9db07dbd2a3a9acf5580892a115c11/packages/utils/src/validators.ts#L147

Reply:
make sense